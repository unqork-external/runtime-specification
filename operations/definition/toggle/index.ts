export * from './toggle.model'
export * from './toggle.options'
