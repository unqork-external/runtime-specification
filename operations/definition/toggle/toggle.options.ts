import { Description, Example, Required } from '@tsed/schema'

export class ToggleOptions {
  @Required()
  @Example('firstNameTextField', 'grid.row(0).col(0)')
  @Description('targetKey refers to the intended target to manipulate.')
  targetKey: string

  @Required()
  @Example('value', 'validation.latest')
  @Description(
    '`property` refers to the intended property to mutate. This can be represented as a JSONPath expression.',
  )
  property: string = 'value'
}
