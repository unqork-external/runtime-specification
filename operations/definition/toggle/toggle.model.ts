import { Description, DiscriminatorValue, Required } from '@tsed/schema'

import { ToggleOptions } from './toggle.options'
import { DisplayName } from '../../../../decorators'
import { Stable } from '../../../../decorators/stability'
import { OperationTypes } from '../../enums/operation-types.enum'
import { Operation } from '../../interface/operations.interface'

@DisplayName('Toggle')
@DiscriminatorValue(OperationTypes.TOGGLE)
@Description('Toggles a boolean property.')
@Stable()
export class ToggleOperation extends Operation<OperationTypes.TOGGLE> {
  public type = OperationTypes.TOGGLE as const

  @Required()
  public options: ToggleOptions = new ToggleOptions()
}
