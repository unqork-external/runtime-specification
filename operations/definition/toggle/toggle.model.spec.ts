import { ToggleOperation } from './toggle.model'
import { generateSchemaAndValidate } from '../../../../utilities'

describe('Operation Model: ToggleOperation Tests', () => {
  let schema
  beforeAll(() => {
    const response = generateSchemaAndValidate(ToggleOperation)
    schema = response.schema
  })

  it('should match this JSON Schema', () => {
    expect(schema).toMatchSnapshot()
  })
})
