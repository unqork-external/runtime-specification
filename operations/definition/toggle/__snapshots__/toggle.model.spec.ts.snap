// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Operation Model: ToggleOperation Tests should match this JSON Schema 1`] = `
{
  "definitions": {
    "ToggleOptions": {
      "properties": {
        "property": {
          "description": "\`property\` refers to the intended property to mutate. This can be represented as a JSONPath expression.",
          "examples": [
            "value",
            "validation.latest",
          ],
          "minLength": 1,
          "type": "string",
        },
        "shouldExecute": {
          "default": true,
          "description": "
    Conditionally execute this operation.
    Can set to false to temporarily disabled when debugging.
  ",
          "type": [
            "string",
            "boolean",
          ],
        },
        "targetKey": {
          "description": "targetKey refers to the intended target to manipulate.",
          "examples": [
            "firstNameTextField",
            "grid.row(0).col(0)",
          ],
          "minLength": 1,
          "type": "string",
        },
      },
      "required": [
        "targetKey",
        "property",
      ],
      "type": "object",
    },
  },
  "description": "Toggles a boolean property.",
  "properties": {
    "creatorSummary": {
      "description": "A detailed summary of the operation",
      "type": "string",
    },
    "name": {
      "description": "Name of the operation",
      "type": "string",
    },
    "options": {
      "$ref": "#/definitions/ToggleOptions",
    },
    "type": {
      "const": "TOGGLE",
      "examples": [
        "TOGGLE",
      ],
      "type": "string",
    },
  },
  "required": [
    "options",
  ],
  "type": "object",
}
`;
